modules/carousel/jetpack-carousel.js

```diff
@@ -1408,6 +1416,37 @@ jQuery(document).ready(function($) {
 		$(this).jp_carousel('open', {start_index: $(this).find('.gallery-item, .tiled-gallery-item').index($(e.target).parents('.gallery-item, .tiled-gallery-item'))});
 	});
 
+	// handle lightbox (single image gallery) for images linking to 'Attachment Page'
+	if ( 1 === Number( jetpackCarouselStrings.single_image_gallery ) ) {
+		// process links that have rel=attachment and img tag inside
+		$( 'a img[data-attachment-id]' ).each(function() {
+			var container = $( this ).parent();
+			var valid = false;
+
+			// if link points to 'Media File' and flag is set allow it
+			if ( $( container ).attr( 'href' ) === $( this ).attr( 'data-orig-file' ) &&
+				1 === Number( jetpackCarouselStrings.single_image_gallery_media_file )
+			) {
+				valid = true;
+			}
+
+			// if link points to 'Attachment Page' allow it
+			if( $( container ).attr( 'href' ) === $( this ).attr( 'data-permalink' ) ) {
+				valid = true;
+			}
+
+			// links to 'Custom URL' or 'Media File' when flag not set are not valid
+			if( ! valid ) {
+				return;
+			}
+
+			// make this node a gallery recognizable by event listener above
+			$( container ).addClass( 'single-image-gallery') ;
+			// method 'open' needs this attribute to distinguish from default WP gallery
+			container.data( 'carousel-extra', { single_image: true } );
```

Comment:
`data-carousel-extra` is required by method `open` to generate the gallery. `single_image` attribute was there. To ensure that this attribute is set for single image galleries as well. 

After taking a closer look it was discovered that field `blog_id` which is set there for "regular" carousel galleries was missing for the ones with single image. It was added, and thus the dummy `single_image` could be removed. 

Fixed in 01ee24d40a00b071a3bc73009b98b93be631086a